### PR TITLE
Fix backend_util.py and test cases to support Aer 0.13

### PR DESCRIPTION
**Pull request by doichanj**
### Summary
This PR is fix of `qiskit/utils/backend_utils.py`  to support Aer 0.13

### Details and comments
Some test cases with noise model fails with Aer 0.13 release because of 
https://github.com/Qiskit/qiskit/blob/5a948c2eb3172a6d073d356d33f793e29e28fd42/qiskit/utils/quantum_instance.py#L299
`is_simulator_backend` always returns `False` with BackendV2.
This function will be deprecated, but I think we have to add support for V2 to pass test cases.

In `test/python/providers/test_fake_backends.py`, ignore running test
when backend is V2.

In `test/python/pulse/test_block.py` test case is removed because
Aer does not have pulse simulator anymore.

### CODE REVIEW

**Comment by woodsp-ibm** [comment]
`backend_utils` module was part of algorithms support logic around use of QuantumInstance (came from Aqua) and all the methods were deprecated and this code will shortly be removed - I would have expected the whole module to removed.  The fake_backends test fix should not add a dependence this file.

Out of curiosity I did a search just to see if either private methods got used elsewhere - seems not - but.... it seems to have been copied elsewhere https://github.com/Qiskit/qiskit/blob/e7b1f5e69a4e5f2e9df599168e5f5c09e61b9353/qiskit/visualization/gate_map.py#L29-L31

Maybe as such having some common function somewhere is useful - I'll leave that to the terra-core team to decide. For me its just `getattr(backend, "version", None)` which could simply be inlined to the test.

File: test/python/providers/test_fake_backends.py
Comment (mtreinish) [inline test/python/providers/test_fake_backends.py]: You can just do:

```suggestion
        if sim.version > 1::
```

The version attribute is the only shared attribute with the `Backend` abstract interface that both `BackendV1` and `BackendV2` inherit from. The reason `_get_backend_interface_version()` existed originally was because the predecessor to `BackendV1`, `BaseBackend`, was not a versioned interface and didn't have a `version` attribute. (it did have a method `version()` if I remember correctly that would return other metadata, so it couldn't be used or overridden for version negotiation).

File: releasenotes/notes/fix_sim_backend-f3971b1ed4d0c4e6.yaml
Comment (mtreinish) [inline releasenotes/notes/fix_sim_backend-f3971b1ed4d0c4e6.yaml]: You can remove these lines as we typically don't need to document test suite changes for our end users as it's not a user facing change.

File: qiskit/utils/backend_utils.py
Comment (woodsp-ibm) [inline qiskit/utils/backend_utils.py:230]: My concern with this would the that the remote simulator "ibmq_qasm_simulator" would now be treated as local simulator wouldn't it. There is logic in run_circuits that dealt with max circuits around whether things were being run locally or not.

Comment (ElePT) [reply qiskit/utils/backend_utils.py:230]: I think that with the upcoming changes in runtime simulators we shouldn't worry about this too much.

File: test/python/circuit/test_controlled_gate.py
Comment (ElePT) [inline test/python/circuit/test_controlled_gate.py:1030]: I understand that this change is to make the code a bit general and not only work for `MCU1Gate`, `MCPhaseGate` and `MCXGate`, but I was wondering, why did this fail with the new Aer?

**Review by ElePT — COMMENTED** [review]
The PR looks good to me, I just had one question to understand how the controlled gate test relates to the rest of changes in the PR, in case this is something to keep in mind even after `backend_utils.py` is removed.

File: test/python/circuit/test_controlled_gate.py
Comment (jakelishman) [reply test/python/circuit/test_controlled_gate.py:1030]: The test discovery attempts to discover every registered subclass of `ControlledGate` (Python makes that information easily accessible), without restriction to the Qiskit standard library, where it would make sense.

We ideally should probably fix the test discovery to limit it only to controlled gates defined in `qiskit.circuit.library.standard_gates`.

Comment (ElePT) [reply test/python/circuit/test_controlled_gate.py:1030]: So you are suggesting changing the current code on line 1017:
```@data(*ControlledGate.__subclasses__())```
with something like: 
```@data([cls for cls in allGates.__dict__.values() if isinstance(cls,type) and issubclass(cls,ControlledGate)])```?

File: qiskit/utils/backend_utils.py
Comment (woodsp-ibm) [reply qiskit/utils/backend_utils.py:230]: 🙈

File: test/python/circuit/test_controlled_gate.py
Comment (jakelishman) [reply test/python/circuit/test_controlled_gate.py:1030]: I'm not a huge fan of the automatic discovery in its current form at all, but given the current file, yeah, something like that would be fine.  Something a little less magic would be
```python
from qiskit.circuit.library.standard_gates import get_standard_gate_name_mapping

standard_controlled_types = [x.base_class for x in get_standard_gate_name_mapping().values() if isinstance(x, ControlledGate)]
```

That would exclude the multi-controlled gates, which imo should be handled specially if they're not going to follow the conventions of standard-library gates.

(_really_, I don't think the mcx etc gates should be in `standard_gates` at all, they should be in `generalized_gates`, but that's an argument for another time)

Comment (ElePT) [reply test/python/circuit/test_controlled_gate.py:1030]: Given that it looks like there is a lot of room for improvement in these tests, let's keep the current fix as-is and consider improving the tests in a follow-up.

**Review by ElePT — APPROVED** [review]
LGTM, let's merge this PR to further unblock the removal of `backend_utils.py`. Thanks a lot @doichanj.

**Pull request merged**
